### ISSUE TITLE
Unable to upgrade Cesium

### ISSUE THREAD

**Issue by M5oul** [YunoHost-Apps/cesium_ynh#10]
We are unable, from the community list, to upgrade Cesium with a:
```bash
sudo yunohost app upgrade cesium
```

To bypass it:
```bash
sudo yunohost app upgrade cesium -u https://github.com/YunoHost-Apps/cesium_ynh
```

Can we find something like the upstream version which change to allow YunoHost to upgrade the app.

**Comment by Jaxom99** [comment]
Same problem here, on a last version YNH, on Raspbian.

**Comment by lapineige** [comment]
Is the community list up-to-date ?

**Comment by Thatoo** [comment]
Is this issue still going on?
Does anyone know why Yunohost doesn't invite us to update Cesium?

**Comment by Thatoo** [comment]
I answer my own question.
`sudo yunohost app upgrade cesium`
is still not working. It tells me that Cesium is already up to date.
However, after 
`sudo yunohost app upgrade cesium -u https://github.com/YunoHost-Apps/cesium_ynh`
Cesium updated from 1.2.x to last version, currently 1.3.11

Is it something we need to change in cesium's yunohost package or in, as lapineige asked, the community list?

**Comment by lapineige** [comment]
I just checked: the community list is up-to-date, and will always be aligned with master: https://github.com/YunoHost/apps/blob/master/community.json#L164

**Comment by Thatoo** [comment]
Aligned with the branch master of https://github.com/YunoHost-Apps/cesium_ynh ?
Where does it take the information? from manifest? Then maybe this PR will solve the issue, https://github.com/YunoHost-Apps/cesium_ynh/pull/12 .

**Comment by lapineige** [comment]
On the precise line that you will see in my previous link.
`"revision": "HEAD"` means `yunohost app upgrade cesium` (or install) will always use https://github.com/YunoHost-Apps/cesium_ynh, at it's current state.
The manifest is not controling cesium app version. It's only some information for Yunohost.

Upgrading the app means upgrading the source, not the number in the manifest.

If I understand well how that install works https://github.com/YunoHost-Apps/cesium_ynh/blob/master/scripts/install, cesiuem_ynh always use the last cesium release available, I suppose.

**Comment by Thatoo** [comment]
> On the precise line that you will see in my previous link.
> "revision": "HEAD" means yunohost app install cesium will always use https://github.com/YunoHost-Apps/cesium_ynh, at it's current state.
> If I understand well how that install works https://github.com/YunoHost-Apps/cesium_ynh/blob/master/scripts/install, cesiuem_ynh always use the last cesium release available, I suppose.

`install` is working and installing the last cesium release. That's ok.

The problem is double about upgrade.
1.  `sudo yunohost app upgrade cesium`
is not responding the same way than 
` sudo yunohost app upgrade cesium -u https://github.com/YunoHost-Apps/cesium_ynh`
Whereas it should for what I understood of your explanations

2. Yunohost doesn't detects available updates of Cesium (in the admin panel)

I guess both problems are linked to each other.

**Comment by lapineige** [comment]
For point 1, does some from the @YunoHost-Apps/apps-group have any clue ?

**Comment by kay0u** [comment]
https://github.com/YunoHost-Apps/cesium_ynh/blob/32ef6859cd3e0d8d2e3215bdcff13687b11c5746/scripts/upgrade#L22

Yunohost can't know that there is an upgrade for the app. For that, you have to use helpers (especially `ynh_setup_source`), and this file: 
https://github.com/YunoHost/example_ynh/blob/master/conf/app.src
And update it each time a new version come for cesium

**Comment by maniackcrudelis** [comment]
Also, the manifest doesn't have any version number for the upstream app nor the package.
And this repo has no update for 2 years, so for YunoHost, there's nothing to update.

**Comment by Thatoo** [comment]
> Also, the manifest doesn't have any version number for the upstream app nor the package.
> And this repo has no update for 2 years, so for YunoHost, there's nothing to update.

I've made two small PR to answer that.
I'll try to look at the rest but it looks more technical...

**Comment by Thatoo** [comment]
@kay0u I have made a third PR trying to follow what you said but I'm not sure I understand it perfectly well. Could you have a quick look and make a comment. If I'm on the good path, I might try to remake this package from the bottom following example_ynh to make it more "yunohost like" style.

**Comment by kay0u** [comment]
A complete remake with the new package standards would indeed be really cool!

Feel free to contact me if you have any questions!

**Comment by M5oul** [comment]
Fixed by #14.
